src/execution_plan/ops/op_conditional_traverse.c

```diff
@@ -267,23 +207,15 @@ static void CondTraverseFree(OpBase *ctx) {
 		op->M = GrB_NULL;
 	}
 
-	if(op->edges) {
-		array_free(op->edges);
-		op->edges = NULL;
-	}
-
 	if(op->ae) {
 		AlgebraicExpression_Free(op->ae);
 		op->ae = NULL;
 	}
 
-	if(op->edgeRelationTypes) {
-		array_free(op->edgeRelationTypes);
-		op->edgeRelationTypes = NULL;
-	}
+	if(op->setEdge) Traverse_FreeEdgeData(&op->edge_data);
```

Comment:
please call Free regardless, remove the `array_clear` from line 177 as it is breakinging abstraction
use a pointer to an allocated struct